messaging/src/main/java/org/axonframework/eventhandling/EventHandlerInvoker.java

```diff
@@ -92,4 +92,13 @@ default <R> void performReset(@Nullable R resetContext) {
             );
         }
     }
+
+    /**
+     * Should be implemented to clear a cache when processing of a segment is stopped. As another instance might start
+     * processing the segment, we can't trust the local cache state anymore for that segment.
+     *
+     * @param segmentId the id of the segment for which the cache needs to be cleared.
+     */
+    default void clearCache(int segmentId) {
```

Comment:
Providing the entire `Segment` instead of the `segmentId` may be better. Although we are not using it right now, I can imagine detecting a segment merge or split (which should be doable based on the given `Segment` and the previous `Segments` an invoker received) may become useful.

Reply:
Done